### PR TITLE
Pull request by GaryOcean428

### PR DESCRIPTION
**Pull request by GaryOcean428**
Please review and merge this change.

### CODE REVIEW

File: execute_script.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import subprocess
+
+subprocess.run(["python", "run.py"])
\ No newline at end of file
```
Comment (GaryOcean428) [inline execute_script.py:3]: This script just runs another script. It would be better to directly run the required script or use a proper task scheduler or script runner tool.

File: llm_switcher.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    def __init__(self, llms, tasks):
+        self.llms = llms
+        self.tasks = tasks
+        self.current_llm = None
+        self.llm_log = pd.DataFrame(columns=['task', 'llm'])
+
+    def select_LLM(self, task):
+        # Use a RandomForestRegressor for simplicity
+        model = RandomForestRegressor()
+        X = pd.DataFrame([vars(llm) for llm in self.llms])
```
Comment (GaryOcean428) [inline llm_switcher.py:17]: The model is trained every time when selecting a LLM which is inefficient. Consider training the model once and reusing it.

File: llm_switcher.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        y = X.pop('performance')
+        model.fit(X, y)
```
Comment (GaryOcean428) [inline llm_switcher.py:19]: You are predicting the performance of the task without considering the cost. Consider using a multi-objective optimization approach.

File: llm_switcher.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        prediction = model.predict([vars(task)])
+        self.current_llm = self.llms[np.argmax(prediction)]
+        return self.current_llm
+
+    def switch_LLM(self, llm):
+        self.current_llm = llm
+
+    def document_LLM(self, task):
```
Comment (GaryOcean428) [inline llm_switcher.py:27]: Appending to a DataFrame in a loop can be very slow. Consider collecting the data in a list and creating the DataFrame at the end.
